twinkle.js

```diff
@@ -446,33 +452,19 @@ Twinkle.load = function () {
 	// Set custom Api-User-Agent header, for server-side logging purposes
 	Morebits.wiki.api.setApiUserAgent('Twinkle (' + mw.config.get('wgWikiID') + ')');
 
-	// Load all the modules in the order that the tabs should appear
-	var twinkleModules = [
-		// User/user talk-related
-		'arv', 'warn', 'block', 'welcome', 'shared', 'talkback',
-		// Deletion
-		'speedy', 'prod', 'xfd', 'image',
-		// Maintenance
-		'protect', 'tag',
-		// Misc. ones last
-		'diff', 'unlink', 'fluff', 'deprod', 'batchdelete', 'batchprotect', 'batchundelete'
-	];
-	// Don't load modules users have disabled
-	var disabledModules = Twinkle.getPref('disabledModules').concat(Twinkle.getPref('disabledSysopModules'));
-	twinkleModules.filter(function(mod) {
-		return disabledModules.indexOf(mod) === -1;
-	}).forEach(function(module) {
-		Twinkle[module]();
-	});
-	Twinkle.config.init(); // Can't turn off
+	Twinkle.disabledModules = Twinkle.getPref('disabledModules').concat(Twinkle.getPref('disabledSysopModules'));
 
-	// Run the initialization callbacks for any custom modules
-	Twinkle.initCallbacks.forEach(function (func) {
-		func();
-	});
-	Twinkle.addInitCallback = function (func) {
-		func();
+	// Redefine addInitCallback so that any modules being loaded now on are directly
+	// initialised rather than added to initCallbacks array
+	Twinkle.addInitCallback = function(func, name) {
+		if (!name || Twinkle.disabledModules.indexOf(name) === -1) {
+			Twinkle[func]();
```

Comment:
Right off the bat, this kills it; should still be `func()`.  As an example, arv adds `{func: Twinkle.arv, name: 'arv'}` to `initCallbacks`.  That passes `twinklearv`, so `Twinkle.twinklearv()` isn't a function (`Twinkle.arv = function twinklearv() {`, but either way, `Twinkle['Twinkle.arv']` isn't a function).

```suggestion
			func();
```

Reply:
Ufoos. Fixed.